R/prior.R

```diff
@@ -90,12 +90,20 @@ epidist_family_prior.default <- function(family, formula, ...) {
 epidist_family_prior.lognormal <- function(family, formula, ...) {
   prior <- brms::prior("normal(1, 1)", class = "Intercept")
   if ("sigma" %in% names(formula$pforms)) {
-    prior <- prior +
-      brms::prior("normal(-0.7, 0.4)", class = "Intercept", dpar = "sigma")
+    # Case with a model on sigma
+    sigma_prior <- brms::prior(
+      "normal(-0.7, 0.4)", class = "Intercept", dpar = "sigma"
+    )
+  } else if ("sigma" %in% names(formula$pfix)) {
```

Comment:
Why did you choose to do this here on a case by case basis and not in epidist_prior (i.e look for anything in the formula that is fixed and remove it from the currently specified prior?

Reply:
`replace_prior` currently errors when you pass it a prior which there is no matching prior for it to overwrite. One possibility would be just to make it less strict and just to have a warning when it's given a prior which doesn't exist. But this should only happen for the user passing in priors, of course for all our code we shouldn't have routine warnings happening.

To remove anything which is fixed it would be like removing anything which is fixed from each of:

```r
model <- epidist_model_prior(data, formula)
family <-  epidist_family_prior(family, formula)
```

I can imagine the code for this being like:

```r
model <- epidist_model_prior(data, formula)
family <-  epidist_family_prior(family, formula)
for(dpar in names(formula$pfix)) { ... }
```

Which I could be happy to move to. It's probably more complicated now but I agree in future it'll be less complicated. It's also mildly unappealing to have e.g. `epidist_model_prior` return the wrong prior (all else equal then it being lower down the stack is better).

Reply:
> One possibility would be just to make it less strict and just to have a warning when it's given a prior which doesn't exist. But this should only happen for the user passing in priors, of course for all our code we shouldn't have routine warnings happening.

I like this suggestion and then run is it constant checks on the complete output as another internal function.

If you feel like this is to much for this PR we can merge and open a new issue but I think we want this generalised ASAP vs pushing down the priority order.

> epidist_model_prior return the wrong prior (all else equal then it being lower down the stack is better).

We don't really expect users to interact with this at all do we so in some sense it doesn't matter?

Reply:
> If you feel like this is to much for this PR we can merge and open a new issue but I think we want this generalised ASAP vs pushing down the priority order.

Yes I feel like a solution along these lines could work but it's not obvious so I'll make a new issue.

> We don't really expect users to interact with this at all do we so in some sense it doesn't matter?

Developers are still a type of user.

Reply:
do you think developers will want to write endless if else statements in their code over it having output that is overwritten ;)

Reply:
Yes it makes me feel productive.

Reply:
#305

Reply:
#305

Reply:
https://github.com/epinowcast/epidist/issues/306